pkg/controller/ippool/controller.go

```diff
@@ -280,6 +280,23 @@ func (h *Handler) DeployAgent(ipPool *networkv1.IPPool, status networkv1.IPPoolS
 		return status, fmt.Errorf("could not find clusternetwork for nad %s", ipPool.Spec.NetworkName)
 	}
 
+	if ipPool.Status.AgentPodRef != nil {
+		pod, err := h.podCache.Get(ipPool.Status.AgentPodRef.Namespace, ipPool.Status.AgentPodRef.Name)
+		if err == nil {
+			if pod.GetUID() == ipPool.Status.AgentPodRef.UID {
+				return status, nil
+			}
+
+			return status, fmt.Errorf("agent pod %s uid mismatch, waiting for removal then redeploy", ipPool.Status.AgentPodRef.Name)
+		} else {
+			if !apierrors.IsNotFound(err) {
+				return status, err
+			}
+
+			logrus.Warningf("(ippool.DeployAgent) agent pod %s missing, redeploying", ipPool.Status.AgentPodRef.Name)
+		}
```

Comment:
NIT: I think we usually check error not equal nil first?

```suggestion
		if err != nil {
			if !apierrors.IsNotFound(err) {
				return status, err
			}

			logrus.Warningf("(ippool.DeployAgent) agent pod %s missing, redeploying", ipPool.Status.AgentPodRef.Name)
		} else {
			if pod.GetUID() == ipPool.Status.AgentPodRef.UID {
				return status, nil
			}

			return status, fmt.Errorf("agent pod %s uid mismatch, waiting for removal then redeploy", ipPool.Status.AgentPodRef.Name)
		}
```

Reply:
It should be like that. Thank for the heads-up!